lib/apis/v3/node.go

```diff
@@ -50,6 +52,15 @@ type NodeSpec struct {
 
 	// OrchRefs for this node.
 	OrchRefs []OrchRef `json:"orchRefs,omitempty" validate:"omitempty"`
+
+	// Wireguard configuration for this node.
+	Wireguard *NodeWireguardSpec `json:"wireguard,omitempty" validate:"omitempty"`
+}
+
+type NodeStatus struct {
+	// PublicKey is the Wireguard public-key for this node.
+	// wireguardPublicKey validates if the string is a valid base64 encoded key.
+	PublicKey string `json:"publicKey,omitempty" validate:"omitempty,wireguardPublicKey"`
```

Comment:
Now that this is in the Status, the field name has lost the "wireguard" association.  So we probably want to either:
-  Add a wireguard section to the status
-  Rename this to be WireguardPublicKey

Both the spec and status now have only a single wireguard field.  Wondering if we should just remove the wireguard struct and put them as top level fields (such as the vxlan fields).  Given the precedence with vxlan I'm tempted to remove the extra level of hierarchy.  WDYT?  Perhaps ping Casey see what he thinks too.

Reply:
>Rename this to be WireguardPublicKey

I like this suggestion.

>Given the precedence with vxlan I'm tempted to remove the extra level of hierarchy.
IMO, indirection is helpful if we have a scope of possible new fields in `NodeWireguardSpec` (e.g. interface ipv6 address or per-node MTU, when available). Not that it cannot exist on its own as VXLAN options, I think it keeps configuration clearer.

I will check with @caseydavenport. If he is of the same opinion, I can post an update merging NodeWireguardSpec into Spec.

Reply:
I haven't been involved in a lot of the UX design to-date, but I think my rough idea would have been:

- Felix always has wireguard on, with an explicit option to _disable_ it in FelixConfig.
- Felix looks for the `Node.Spec.Wireguard != nil` to indicate that it should do some wg stuff
- The `Node.Spec.Wireguard` section includes whatever configuration we might want to expose, including MTU, etc.


I get that this is different from VXLAN / IPIP in some ways, but I think is a better UX. Namely because it doesn't make the user go back and forth between FelixConfiguration and the Node resource somewhat arbitrarily. WDYT?

Reply:
Otherwise, I don't have too strong an opinion. This object is already sorta inconsistent with itself. I think a separate wireguard subsection is probably a good idea, in case we want to add other config options there (although it looks like this PR is putting them all in FelixConfig, hence my above comment)

Reply:
Thanks Casey. I have kept the wireguard section in Spec and renamed `PublicKey` to `WireguardPublicKey` now.